PcapDotNet/src/PcapDotNet.Packets/Dhcp/DhcpDatagram.cs

```diff
@@ -0,0 +1,368 @@
+        /// </summary>
+        public IpV4Address NextServerIpAddress
+        {
+            get { return ReadIpV4Address(Offset.SiAddr, Endianity.Big); }
+        }
+
+        /// <summary>
+        /// RFC 2131.
+        /// Relay agent IP address, used in booting via a relay agent.
+        /// </summary>
+        public IpV4Address RelayAgentIpAddress
+        {
+            get { return ReadIpV4Address(Offset.GiAddr, Endianity.Big); }
+        }
+
+        /// <summary>
+        /// RFC 2131.
+        /// Client hardware address.
+        /// </summary>
+        public DataSegment ClientHardwareAddress
+        {
+            get { return this.Subsegment(Offset.ChAddr, 16); }
+        }
+
+        /// <summary>
+        /// Client MAC address
+        /// </summary>
+        public MacAddress ClientMacAddress
+        {
+            get { return new MacAddress(ReadUInt48(Offset.ChAddr, Endianity.Big)); }
+        }
+
+        /// <summary>
+        /// RFC 2131.
+        /// Optional server host name
+        /// </summary>
+        public string ServerHostName
+        {
+            get
+            {
+                ParseServerHostName();
+                return _serverHostName;
+            }
+        }
+
+        /// <summary>
+        /// RFC 2131.
+        /// Boot file name; "generic" name or null in DHCPDISCOVER, fully qualified directory-path name in DHCPOFFER.
+        /// </summary>
+        public string BootFileName
+        {
+            get
+            {
+                ParseBootFileName();
+                return _bootFileName;
+            }
+        }
+
+        /// <summary>
+        /// RFC 2131.
+        /// true if the magic dhcp-cookie is set
+        /// </summary>
+        public bool IsDhcp
+        {
+            get
+            {
+                if (Length >= Offset.Options + 4)
+                {
+                    return ReadInt(Offset.Options, Endianity.Big) == DHCP_MAGIC_COOKIE;
+                }
+                else
+                {
+                    return false;
+                }
+            }
+        }
+
+        /// <summary>
+        /// RFC 2131.
+        /// Optional parameters field
+        /// </summary>
+        public IReadOnlyCollection<DhcpOption> Options
+        {
+            get
+            {
+                ParseOptions();
+                return _options;
+            }
+        }
+
+        internal DhcpDatagram(byte[] buffer, int offset, int length) : base(buffer, offset, length)
+        {
+        }
+
+        private void ParseServerHostName()
+        {
+            if (_serverHostName == null)
+            {
+                //at the moment we only interpret server host name as sname and ignore options
+                byte[] byteServerHostName = ReadBytes(Offset.Sname, 64);
+                _serverHostName = Encoding.ASCII.GetString(byteServerHostName).TrimEnd('\0');
```

Comment:
Regarding TrimEnd(), what happens if after the null character there's a non null character?